tests/rebin_test.py

```diff
@@ -274,3 +274,56 @@ def test_uncal_spectrum_counts(self, old_spec_float):
             label="rebinned",
         )
         plt.show()
+
+
+@pytest.mark.parametrize(
+    "edges2, counts2, method, include_overflows",
+    [
+        (
+            np.linspace(2, 9, num=8),
+            np.array([3000, 1000, 1000, 1000, 1000, 1000, 2000]),
+            "interpolation",
+            True,
+        ),
+        (
+            np.linspace(2, 9, num=8),
+            np.array([1000, 1000, 1000, 1000, 1000, 1000, 1000]),
+            "interpolation",
+            False,
+        ),
+        (
+            np.linspace(2, 9, num=8),
+            np.array([3000, 1000, 1000, 1000, 1000, 1000, 2000]),
+            "listmode",
+            True,
+        ),
+        (
+            np.linspace(2, 9, num=8),
+            np.array([1000, 1000, 1000, 1000, 1000, 1000, 1000]),
+            "listmode",
+            False,
+        ),
```

Comment:
should listmode give the same results as interpolation here?

Reply:
Yes, because in this special case the new bins (2, 3, ... 9) are an exact subset of the original bins (0, 1, 2, ..., 9, 10). So when `listmode=True`, any events sampled from an original bin end up in exactly the same bin, so the randomness does not change the bin contents.